cassiopeia/preprocess/utilities.py

```diff
@@ -451,36 +458,242 @@ def convert_alleletable_to_character_matrix(
                     continue
 
                 if state == "NONE" or "None" in state:
-                    character_strings[sample].append("0")
+                    character_strings[sample].append(0)
                 else:
                     if state in allele_counter[c]:
                         character_strings[sample].append(
-                            str(allele_counter[c][state])
+                            allele_counter[c][state]
                         )
                     else:
                         # if this is the first time we're seeing the state for this character,
                         # add a new entry to the allele_counter
                         allele_counter[c][state] = len(allele_counter[c]) + 1
                         character_strings[sample].append(
-                            str(allele_counter[c][state])
+                            allele_counter[c][state]
                         )
 
+                        indel_to_charstate[i][len(allele_counter[c])] = state
+
                         # add a new entry to the character's probability map
                         if mutation_priors is not None:
                             prob = np.mean(mutation_priors.loc[state, "freq"])
-                            prior_probs[i][
-                                str(len(allele_counter[c]))
-                            ] = float(prob)
-                            indel_to_charstate[i][
-                                str(len(allele_counter[c]))
-                            ] = state
+                            prior_probs[i][len(allele_counter[c])] = float(prob)
+
             else:
                 character_strings[sample].append(missing_data_state)
 
     character_matrix = pd.DataFrame.from_dict(
         character_strings,
         orient="index",
-        columns=[f"r{i}" for i in range(1, len(intbc_uniq)+1)],
+        columns=[f"r{i}" for i in range(1, len(intbc_uniq) + 1)],
     )
 
     return character_matrix, prior_probs, indel_to_charstate
+
+
+def convert_alleletable_to_lineage_profile(allele_table) -> pd.DataFrame:
+    """Converts an alleletable to a lineage profile.
+
+    Takes in an allele table that summarizes the indels observed at individual
+    cellBC-intBC pairs and produces a lineage profile, which essentially is a 
+    pivot table over the cellBC / intBCs. Conceptually, these lineage profiles
+    are identical to character matrices, only the values in the matrix are the
+    actual indel identities.
+
+    Args:
+        allele_table: AlleleTable.
+
+    Returns:
+        An NxM lineage profile.
+    """
+
+    cutsites = [
+        column
+        for column in allele_table.columns
+        if bool(re.search(r"r\d", column))
```

Comment:
This logic seems to be duplicated from before, maybe we should factor it out into a function `def cutsites(...)`.